### ISSUE TITLE
ZFS 0.6.5.8-1.el7 modules not loading on latest RHEL 7.3 Kernel  3.10.0-514.el7

### ISSUE THREAD

**Issue by BonganiMalalane** [zfsonlinux/zfs#5446]
Recently upgraded my RHEL 7 systems to latest minor release (7.3) and kernel (3.10.0-514).
Rebooted the systems and noticed none of zpools imported. When manually importing pools it asked me to run a "modprobe zfs", which I did and it failed not finding the modules.
Reinstalled kernel, kmod, and in fact the entire zfs stack and that didn't work. 
Rolled back to kernel (3.10.0-327.36.3.el7) and it works. Seems to be kernel dependant and runs on all the (3.10.0-327.x) versions.

**Comment by behlendorf** [comment]
Thanks for filing the issue.  This is a duplicate of #5015 and #5377 and was caused by a change RedHat made to the kernel.  For the moment I'd suggest switching to the DKMS packages until we can past packages for 7.3 and newer kernels in a separate repo.

**Comment by dmaziuk** [comment]
Brian, do you have an ETA guesstimate for the new repo? DKMS is a pain and switching back to it temporarily is not something I'm looking forward to. So I wonder if we're better off just waiting a bit.

Right now there's a chicken and egg situation with centos: elrepo modules (drbd, nvidia) are already built for the newer kernels and cannot be updated. Installing 7.3 updates via ``cr`` kills zfs. I don't have to enable cr on my (other) zfs hosts so until centos 7.3 is officially out I can just ignore elrepo errors.

**Comment by behlendorf** [comment]
@dmaziuk could you summarize how this transition handled for other elrepo modules (drbd, nvidia).  I haven't looked closely at those packages, but  depending on what symbols are needed they may be usable on both 7.2 and 7.3 kernels.   Did they update their existing repository with these new potentially incompatible packages or did they create a new repository?

As for an ETA I should be able to get the packages built by next week.

**Comment by dmaziuk** [comment]
They did not create a new repo, they made in-place updates that remove ``drbd.ko`` from older kernels:
```
[root@herring modules]# ls
3.10.0-327.36.1.el7.x86_64  3.10.0-327.36.2.el7.x86_64	3.10.0-327.36.3.el7.x86_64  3.10.0-514.2.2.el7.x86_64  3.10.0-514.el7.x86_64
[root@herring modules]# find . -name \*drbd\*
./3.10.0-514.el7.x86_64/extra/drbd84
./3.10.0-514.el7.x86_64/extra/drbd84/drbd.ko
./3.10.0-514.2.2.el7.x86_64/weak-updates/drbd84
./3.10.0-514.2.2.el7.x86_64/weak-updates/drbd84/drbd.ko
```
My guess is there is no good way to handle it, perhaps do what they did and just tell people who want to use older kernels to not update and/or switch to dkms?

**Comment by behlendorf** [comment]
Actually it looks like they created a 7.3.1611 repository to replace the 7.2.1511 repository.  The existing 7 repository is a symlink which currently points to the 7.2.1511 repository but they'll presumably update that when they release 7.3.  I'll see about doing the same thing for the ZFS repos.

```
http://mirror.centos.org/centos/7.2.1511/
http://mirror.centos.org/centos/7.3.1611/
http://mirror.centos.org/centos/7 -> http://mirror.centos.org/centos/7.2.1511/
```

**Comment by behlendorf** [comment]
Official packages have been published for RHEL/CentOS 7.3.

https://github.com/zfsonlinux/zfs/wiki/RHEL-&-CentOS#rhelcentos-73-kmod-packages

**Comment by ionutz22** [comment]
Hi,

Using the URL(baseurl) from documentation works fine.
The kernel version is 3.10.0-514 not 2.6.32-514.
Also please note CentOS 7.3.1611 was released today.

Thank you.

**Comment by behlendorf** [comment]
> The kernel version is 3.10.0-514 not 2.6.32-514.

Fixed.  Thanks!

> Also please note CentOS 7.3.1611 was released today.

So [it was](http://news.softpedia.com/news/centos-linux-7-1611-released-it-s-derived-from-red-hat-enterprise-linux-7-3-510911.shtml!)  Thanks for the heads up, I'll update the zfs-release package accordingly.

**Comment by ionutz22** [comment]
Just a note, if you've already installed zfs previously (0.6.5.8-1.el7.centos) you'll have to remove it then install again from the 7.3 kmod repo.

yum remove spl kmod-spl kmod-zfs zfs

yum install zfs

this will install: 0.6.5.8-1.el7_3.centos + deps.

**Comment by behlendorf** [comment]
@ionutz22 yes good point.  I've updated the documentation accordingly.

**Comment by dmaziuk** [comment]
This has an unsdesirable side-effect: it renames ``/etc/zfs/zed.d/zed.rc`` to ``zed.rc.rpmsave`` and then installs a clean ``zed.rc``. If you uncommented ``ZED_EMAIL_ADDR`` before, you need to do it again.

**Comment by ionutz22** [comment]
indeed, when you remove packages it will create .rpmsave ... usually it notifies this on the screen.
to avoid this you can just adjust the repository and update - in this case a .rpmnew file will be created.
